chain-extensions/impls/dapps-staking/src/lib.rs

```diff
@@ -0,0 +1,476 @@
+            }
+
+            DappsStakingFunc::EraStaked => {
+                let arg: u32 = env.read_as()?;
+                let era_info = pallet_dapps_staking::GeneralEraInfo::<T>::get(arg)
+                    .ok_or(DispatchError::Other(
+                    "[ChainExtension] DappsStakingExtension EraStaked general_era_info call failed",
+                ));
+                let staked_amount = era_info.map_or(Zero::zero(), |r| r.staked);
+                env.write(&staked_amount.encode(), false, None)
+                    .map_err(|_| {
+                        DispatchError::Other(
+                            "[ChainExtension] DappsStakingExtension EraStaked failed to write result",
+                        )
+                    })?;
+            }
+
+            DappsStakingFunc::StakedAmount => {
+                let staker: T::AccountId = env.read_as()?;
+                let ledger = pallet_dapps_staking::Ledger::<T>::get(&staker);
+                env.write(&ledger.locked.encode(), false, None)
+                    .map_err(|_| {
+                        DispatchError::Other(
+                            "[ChainExtension] DappsStakingExtension StakedAmount failed to write result",
+                        )
+                    })?;
+            }
+
+            DappsStakingFunc::StakedAmountOnContract => {
+                let args: DappsStakingAccountInput = env.read_as()?;
+                let staker = T::AccountId::decode(&mut args.staker.as_ref()).unwrap();
+                let contract = Self::decode_smart_contract(args.contract)?;
+                let staking_info =
+                    pallet_dapps_staking::GeneralStakerInfo::<T>::get(&staker, &contract);
+                let staked_amount = staking_info.latest_staked_value();
+                env.write(&staked_amount.encode(), false, None)
+                    .map_err(|_| {
+                        DispatchError::Other(
+                            "[ChainExtension] DappsStakingExtension StakedAmountOnContract failed to write result",
+                        )
+                    })?;
+            }
+
+            DappsStakingFunc::ReadContractStake => {
+                let contract_bytes: [u8; 32] = env.read_as()?;
+                let contract = Self::decode_smart_contract(contract_bytes)?;
+                let current_era = pallet_dapps_staking::CurrentEra::<T>::get();
+                let staking_info =
+                    pallet_dapps_staking::Pallet::<T>::contract_stake_info(&contract, current_era)
+                        .unwrap_or_default();
+                let total = TryInto::<u128>::try_into(staking_info.total).unwrap_or(0);
+                env.write(&total.encode(), false, None).map_err(|_| {
+                    DispatchError::Other(
+                        "[ChainExtension] DappsStakingExtension ReadContractStake failed to write result",
+                    )
+                })?;
+            }
+
+            DappsStakingFunc::Register => {
```

Comment:
We will deprecate this call soon (will be made root-only) so I'd suggest not adding it at all.